app/src/main/java/org/openobservatory/ooniprobe/activity/RunTestActivity.java

```diff
@@ -0,0 +1,236 @@
+package org.openobservatory.ooniprobe.activity;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.AppCompatButton;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openobservatory.ooniprobe.BuildConfig;
+import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.adapter.UrlListAdapter;
+import org.openobservatory.ooniprobe.data.TestData;
+import org.openobservatory.ooniprobe.model.NetworkMeasurement;
+import org.openobservatory.ooniprobe.utils.Alert;
+
+import java.util.ArrayList;
+import java.util.Observable;
+import java.util.Observer;
+import java.util.Set;
+
+import uk.co.chrisjenx.calligraphy.CalligraphyConfig;
+import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;
+
+public class RunTestActivity extends AppCompatActivity implements Observer {
+    private RecyclerView testUrlList;
+    private UrlListAdapter mUrlListAdapter;
+    private static ImageView testImage;
+    private static AppCompatButton runButton;
+    private static TextView title, testTitle, urls;
+    private ProgressBar test_progress;
+    private static String test_name;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_run_test);
+
+        // Get the intent that started this activity
+        Intent intent = getIntent();
+        Uri data = intent.getData();
+        if (Intent.ACTION_VIEW.equals(intent.getAction())) {
+            Uri uri = intent.getData();
+            String mv = uri.getQueryParameter("mv");
+            if (mv != null){
+                if (versionCompare(BuildConfig.VERSION_NAME, mv) >= 0) {
+                    Set<String> parameters = uri.getQueryParameterNames();
+                    String tn = uri.getQueryParameter("tn");
+                    String ta = uri.getQueryParameter("ta");
+                    String td = uri.getQueryParameter("td");
+                    String test = NetworkMeasurement.getTestName(this, tn);
+                    if (test.length() > 0){
+                        test_name = tn;
+                        configureScreen(td, ta);
+                    }
+                    else {
+                        Alert.alertDialog(this, getString(R.string.invalid_parameter), getString(R.string.test_name) +  " : " + tn);
```

Comment:
Does this mean that we don't redirect the user to the store if the version is too small?